packages/foreman/rubygem-fog-google/rubygem-fog-google.spec

```diff
@@ -1,77 +1,118 @@
+BuildRequires: %{?scl_prefix_ruby}ruby < 3
 BuildRequires: %{?scl_prefix_ruby}rubygems-devel
-BuildRequires: %{?scl_prefix_ruby}rubygems
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
-%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}}
-
-%define gembuilddir %{buildroot}%{gem_dir}
+# end specfile generated dependencies
 
 %description
 This library can be used as a module for `fog` or as standalone provider to
-use the Google in applications.
+use the Google Cloud in applications.
+
 
 %package doc
-BuildArch:  noarch
-Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
-%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}-doc}
-Summary:    Documentation for rubygem-%{gem_name}
+Summary: Documentation for %{pkg_name}
+Group: Documentation
+Requires: %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+BuildArch: noarch
 
 %description doc
-This package contains documentation for rubygem-%{gem_name}.
+Documentation for %{pkg_name}.
 
 %prep
-%setup -n %{pkg_name}-%{version} -T -c
-%{?scl:scl enable %{scl} - <<EOF}
-%gem_install -n %{SOURCE0}
+%{?scl:scl enable %{scl} - << \EOF}
+gem unpack %{SOURCE0}
+%{?scl:EOF}
+
+%setup -q -D -T -n  %{gem_name}-%{version}
+
+%{?scl:scl enable %{scl} - << \EOF}
+gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
 %{?scl:EOF}
 
 %build
+# Create the gem as gem install only works on a gem file
+%{?scl:scl enable %{scl} - << \EOF}
+gem build %{gem_name}.gemspec
+%{?scl:EOF}
+
+# %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
+# by default, so that we can move it into the buildroot in %%install
+%{?scl:scl enable %{scl} - << \EOF}
+%gem_install
+%{?scl:EOF}
 
 %install
 mkdir -p %{buildroot}%{gem_dir}
-cp -a .%{gem_dir}/* %{buildroot}%{gem_dir}
+cp -pa .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
 
 %files
 %dir %{gem_instdir}
-%{gem_instdir}/lib
+%{gem_instdir}/.codecov.yml
+%{gem_instdir}/.editorconfig
+%{gem_instdir}/.fog.example
+%exclude %{gem_instdir}/.gitignore
+%exclude %{gem_instdir}/.hound.yml
+%exclude %{gem_instdir}/.rubocop.yml
+%{gem_instdir}/.ruby-gemset
+%exclude %{gem_instdir}/.travis.yml
+%license %{gem_instdir}/LICENSE.md
+%{gem_instdir}/MIGRATING.md
+%{gem_instdir}/ci
+%{gem_libdir}
+%{gem_instdir}/tasks
 %exclude %{gem_cache}
 %{gem_spec}
-%{gem_instdir}/LICENSE.md
-%exclude %{gem_instdir}/.*
 
 %files doc
 %doc %{gem_docdir}
-%{gem_instdir}/CONTRIBUTING.md
-%{gem_instdir}/CONTRIBUTORS.md
-%{gem_instdir}/README.md
+%doc %{gem_instdir}/CHANGELOG.md
+%doc %{gem_instdir}/CONTRIBUTING.md
+%doc %{gem_instdir}/CONTRIBUTORS.md
+%{gem_instdir}/Gemfile
+%doc %{gem_instdir}/README.md
+%{gem_instdir}/Rakefile
 %{gem_instdir}/examples
+%{gem_instdir}/fog-google.gemspec
 %{gem_instdir}/test
-%{gem_instdir}/tests
-%{gem_instdir}/gemfiles
-%{gem_instdir}/Gemfile*
-%{gem_instdir}/Rakefile
-%exclude %{gem_instdir}/%{gem_name}.gemspec
 
 %changelog
+* Tue Mar 12 2019 kgaikwad <kavitagaikwad103@gmail.com> 1.8.2-1
```

Comment:
This is probably a side effect of running the script twice. One changelog should suffice :)